firewood/src/merkle/proof.rs

```diff
@@ -640,6 +732,9 @@ fn unset_internal<K: AsRef<[u8]>, S: ShaleStore<Node> + Send + Sync, T: BinarySe
                 return Ok(false);
             }
 
+            let node = n.chd();
+            let cur_key = n.path.clone().into_inner();
```

Comment:
This looks wrong... I think I just moved this code. I think we can borrow the path here.